src/scripts/main.js

```diff
@@ -1,5 +1,43 @@
 'use strict';
 
-document.addEventListener('click', (e) => {
-  // write code here
+const HALF = 2;
+const wall = document.querySelector('.wall');
+const spider = document.querySelector('.spider');
+
+if (!wall) {
+  throw new Error('Wall element not found');
+}
+
+if (!spider) {
+  throw new Error('Spider element not found');
+}
+
+wall.addEventListener('click', (e) => {
+  const wallCoords = wall.getBoundingClientRect();
+
+  const spiderCoords = {
+    top:
+      e.clientY - wallCoords.top - wall.clientTop - spider.clientHeight / HALF,
```

Comment:
The calculation for `spiderCoords.top` is correct, but ensure that `wall.clientTop` and `spider.clientHeight` are defined and return expected values. If either is undefined, it could lead to unexpected behavior.